lib/internal/Magento/Framework/Url.php

```diff
@@ -488,6 +485,26 @@ public function getBaseUrl($params = [])
         return $result;
     }
 
+    /**
+     * Set Requested scope, or custom scope if it sets on params.
+     *
+     * @param array|null $params
+     * @return \Magento\Framework\Url\ScopeInterface
+     */
+    private function setRequestScope($params = null)
```

Comment:
This method is not following CQRS principle and therefore is too complicated. Setter should not be used as a getter.